gentle_gnomes/src/azavea.py

```diff
@@ -57,18 +62,65 @@ def get_nearest_city(self, lat: float, lon: float, limit: int = 1, **kwargs) ->
             city = cities['features'][0]
             return City(city['properties']['name'], city['properties']['admin'], city['id'])
 
-    def get_scenarios(self, **kwargs) -> List:
+    def get_scenarios(self, **kwargs) -> t.List:
         """Return all available scenarios."""
         return self._get('/scenario', **kwargs)
 
-    def get_indicators(self, **kwargs) -> Dict:
+    def get_indicators(self, **kwargs) -> t.Dict:
         """Return the full list of indicators."""
         return self._get('/indicator', **kwargs)
 
-    def get_indicator_details(self, indicator: str, **kwargs) -> Dict:
+    def get_indicator_details(self, indicator: str, **kwargs) -> t.Dict:
         """Return the description and parameters of a specified indicator."""
         return self._get(f'/indicator/{indicator}', **kwargs)
 
-    def get_indicator_data(self, city: int, scenario: str, indicator: str, **kwargs) -> Dict:
+    def get_indicator_data(self, city: int, scenario: str, indicator: str, **kwargs) -> t.Dict:
         """Return derived climate indicator data for the requested indicator."""
         return self._get(f'/climate-data/{city}/{scenario}/indicator/{indicator}', **kwargs)
+
+
+class AsyncProxy:
+
+    def __init__(self):
+        self._pending = []  # [(url, kwargs), ...]
+
+    async def _process_requests(self, *requests: t.Tuple[str, dict]) -> t.Tuple[t.Union[dict, list]]:
+        session = aiohttp.ClientSession()
+
+        async def make_request(request):
+            url, kwargs = request
+            async with session.get(url, **kwargs) as response:
+                response.raise_for_status()
```

Comment:
Setting `raise_for_status=True` in the session would remove the need for calling the function. But you should still respect the request's kwargs to allow overriding the session's config.

Reply:
Good point. Not enforcing it means handling the the json decode error, which isn't a big deal.